status-im/status-react
# chore: remove mock data from syncing

fixes: https://github.com/status-im/status-mobile/issues/15142

This pr removes the mock data on the syncing pages and adjusts the UI on that page.

Also adds component spec tests to settings list component which were missing.
Updates settings list component to have variant with Status tags in it. (Have requested designers to update Figma file with this variant too)

To Test:
Sync some devices and see the sync device list - page should look correct as in the designs at:


https://www.figma.com/file/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?type=design&node-id=675-177778&t=JYlgu6dKNjwNthjM-4

This page can be navigated to by: Profile -> Syncing

Functionality should work, except for Pair & Unpair device which will be a follow up issue

Also sign in with syncing (onboarding flow) should be checked
https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?type=design&node-id=3068-333697&t=ewCQ8tWKGQB1KBqV-4


Also for this pr I had to adjust `Quo/Title` component to remove the preset margin to the component and instead allow it to be passed as a prop via a `container-style` prop. As a result I had to add margin to the pages using this: Enable Biometrics, Enable Notifications, Welcome To Web 3, Syncing In Progress and Syncing Results page. I verified all of these using Figma and everything looks correct in terms of positioning and padding 👍
